tests/Stripe/Service/AbstractServiceTest.php

```diff
@@ -57,4 +57,22 @@ public function testRetrieveThrowsIfIdNullIsWhitespace()
 
         $this->service->retrieve(' ');
     }
+
+    public function testFormatParams()
+    {
+        $result = \Stripe\Service\AbstractService::formatParams(['foo' => null]);
+        static::assertTrue('' === $result['foo']);
+        static::assertTrue(null !== $result['foo']);
+
+        $result = \Stripe\Service\AbstractService::formatParams(['foo' => ['bar' => null, 'baz' => 1]]);
+        static::assertTrue('' === $result['foo']['bar']);
+        static::assertTrue(null !== $result['foo']['bar']);
+        static::assertTrue(1 === $result['foo']['baz']);
+
+        $result = \Stripe\Service\AbstractService::formatParams(['foo' => ["bar", null, null, "baz"]]);
```

Comment:
What about passing 
```
[
  'email' => null,
  'invoice_settings' => [
    'default_payment_method' => 'null',
  ],
]
```

As in, this method didn't seem recurring or able to handle non top-level params

Reply:
I don't quite follow -- this PR adds a recursive call https://github.com/stripe/stripe-php/pull/933/files#diff-d6b577df0148860a08e6dfb3025bd0d0R52

and this test case covers non top-level params -- everything is nested one level under "foo".